src/main/src/main/java/org/geoserver/catalog/util/ReaderDimensionsAccessor.java

```diff
@@ -171,24 +185,28 @@ public boolean hasTime() throws IOException {
         }
 
         TreeSet<Object> result = null;
-        if (reader instanceof StructuredGridCoverage2DReader) {
-            StructuredGridCoverage2DReader sr = (StructuredGridCoverage2DReader) reader;
-            result = getDimensionValuesInRange("time", range, maxEntries, sr);
-        }
-
-        // if we got here, the optimization did not work, do the normal path
-        if (result == null) {
-            result = new TreeSet<>(TEMPORAL_COMPARATOR);
-            TreeSet<Object> fullDomain = getTimeDomain();
+        if (dimensionInfo != null && dimensionInfo.getFixedValues() != null) {
+            result = getFixedValueRange(dimensionInfo.getFixedValues());
```

Comment:
I see this method is still used. Byt is just returning a TreeSet of String.... Can you explain why we are using it?